lib/mongoid/contextual/aggregable/mongo.rb

```diff
@@ -124,7 +124,7 @@ def sum(field = nil)
         def pipeline(field)
           db_field = "$#{database_field_name(field)}"
           pipeline = []
-          pipeline << { "$match" => criteria.nin(field => nil).selector }
+          pipeline << { "$match" => { field => criteria.selector[field.to_s] } }
```

Comment:
@durran Shouldn't we be setting the $match operator to the criteria selector, not to a selector finding all documents in which the field exists?

In other words, wouldn't this:

```ruby
account.campaigns.where(name: /Untitled (\d+)/).max(:name)
```

return the max of name values of campaigns matching the where criteria, not the max name value across all documents?
I took the example from the JIRA ticket: https://jira.mongodb.org/browse/MONGOID-4119